src/ott/solvers/linear/continuous_barycenter.py

```diff
@@ -174,52 +238,77 @@ def init_state(
       ))
     else:
       errors = None
+    state = FreeBarycenterState(
+        x=x,
+        a=a,
+        costs=-jnp.ones((num_iter,)),
+        linear_convergence=-jnp.ones((num_iter,)),
+        errors=errors
+    )
+    abstract_tree = jax.eval_shape(
+        functools.partial(state.update, store_errors=self.store_inner_errors),
+        0, bar_prob, self.linear_solver
+    )
+    tree = jax.tree.map(jnp.zeros_like, abstract_tree)
```

Comment:
This is a bit wasteful, you can just do `linear_outputs = jax.tree.map(jnp.zeros_like, abstract_tree.linear_outputs)`.

Reply:
indeed! good point.